networkdb/cluster.go

```diff
@@ -107,7 +107,6 @@ func (nDB *NetworkDB) clusterInit() error {
 		config.BindPort = nDB.config.BindPort
 	}
 
-	config.ProtocolVersion = memberlist.ProtocolVersionMax
```

Comment:
If in the future they will update that value, but for us will break only if they will put 6. The logic in the memberlist is such that the current version of the protocol running has to be within the range of the entire cluster (does not seem to affect the behavior of the process itself).
For 17.05 the range is [1-3], after the memberlist bump the range will be [1-5].
The bug is due to that line that was setting the current to 5 making the new nodes 17.06 outside of the range of [1-3] of the 17.05 or previous.
I think would be rare that we will have to deal with such a breakage again, but if you guys agree that is more explicit and clear, I can set it to that value